src/zone-patch/index.js

```diff
@@ -44,51 +44,27 @@ function wrapDescribeInZone(describeBody) {
 // Create a proxy zone in which to run `test` blocks so that the tests function
 // can retroactively install different zones.
 const testProxyZone = ambientZone.fork(new ProxyZoneSpec());
-function wrapTestInZone(testBody, eachArgs) {
+function wrapTestInZone(testBody) {
   if (testBody === undefined) {
     return;
   }
 
-  if (!eachArgs || eachArgs.length === 0 || eachArgs[0].length === 0) {
-    // If we are not handling `test.each`, then the parameter of `testBody`
-    // will be 0 or 1, if it is 1, then we need to return a function with
-    // done parameter
+  const wrappedFunc = function() {
+    return testProxyZone.run(testBody, null, arguments);
+  };
+  try {
+    Object.defineProperty(wrappedFunc, 'length', {
+      configurable: true,
+      writable: true,
+      enumerable: false
+    });
+    wrappedFunc.length = testBody.length;
+  } catch (e) {
     return testBody.length === 0
       ? () => testProxyZone.run(testBody, null)
       : done => testProxyZone.run(testBody, null, [done]);
-  } else {
-    // Dynamically create a Function to contain the same length
-    // of the parameters as the testBody
-    // For example:
-    // ```
-    // test.each([[1, 2]])('test.each', (arg1, arg2) => {});
-    // ```
-    // In this case we need to return a function like this
-    // ```
-    // return function(arg1, arg2) {
-    //   return testProxyZone.run(testBody, null, [arg1, arg2]);
-    // }
-    // ```
-    const len = eachArgs[0].length;
-    const args = [];
-    let argString = '';
-    for (let i = 0; i < len; i++) {
-      args.push('arg' + i);
-      argString += 'arg' + i;
-      if (i !== len - 1) {
-        argString += ', ';
-      }
-    }
-    args.push('testBody');
-    args.push('testProxyZone');
-    if (len < testBody.length) {
-      args.push('done');
-      argString += ', done';
-    }
-    const funcBody = `
-      return testProxyZone.run(testBody, null, [${argString}])`;
-    return new Function(args, funcBody);
```

Comment:
Is the previous fix not required anymore?

Reply:
@wtho , yeah, the reason the previous version is to make sure to return a function with the same length as the test function.
For example,

```
test.each([[1,2]])((arg1, arg2) => {
  expect(arg1).toBe(1);
  expect(arg2).toBe(2);
})
```

So the test function is  `(arg1, arg2) => {...}`, and we need to wrap the function and return a new function to let it run in the `ProxyZone`.
But if we return the function like this
```
return function() {
  return originalTestBody.apply(this, arguments);
}
```
the returned `function.length` will be `0`, not `2` in this example, and `jest` need the function.length to check whether it has `done()` or not.

So in the previous version, I returned a dynamic function with the correct length.
But the previous commit didn't consider the `template literal case`, so the bug occurs.

And in fact, I don't need such complex logic, what I need is just return a function with dynamic length.

```
const wrappedFunc = function() {return originalFunc.apply(this, arguments);
wrappedFunc.length = originalFunc.length;
return wrappedFunc;
``` 

Will handle all the cases.